contracts/libraries/PercentageLib.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import { GenericTypes } from "./DataTypes.sol";
+
+/// @title PercentageLib
+/// @notice Library with helper functions for percentage operations
+library PercentageLib {
+    using GenericTypes for GenericTypes.Percentage;
+
+    /**
+     * @notice This function calculates the percentage of a given amount using a Percentage struct.
+     * @dev This function performs a simple percentage calculation.
+     *      It multiplies the amount by the percentage number and divides the result by 10 raised to the power of decimal places.
+     *      It reverts if the percentage exceeds 100.
+     * @param _amount The amount for which the percentage is calculated.
+     * @param _percentage The percentage to be calculated, represented as a Percentage struct.
+     * @return uint256 The calculated percentage value.
+     * 
+     * Examples:
+     * - If _amount is 1000 and _percentage is {percentageNumber: 20, decimalPlaces: 0}, it means (20% of 1000).
+     * - If _amount is 1000 and _percentage is {percentageNumber: 2345, decimalPlaces: 2}, it means (23.45% of 1000).
+     * - If _amount is 1000 and _percentage is {percentageNumber: 56789, decimalPlaces: 3}, it means (56.789% of 1000).
+     */
+    function calcPercentage(uint256 _amount, GenericTypes.Percentage memory _percentage) internal pure returns (uint256) {
+        uint256 divisor = 10 ** _percentage.decimalPlaces;
```

Comment:
There are two errors, one leads to an overflow in the core contract, and the second one calculates the wrong percentage

Error 1 - `Divisor` should be `100 ** _percentage.decimalPlaces;` instead of `10 **_percentage.decimalPlaces;`

Error 2 - When `decimalPlace` is 0 the `divisor` becomes `10 ** 0 = 1` which means if I passed `20` and `0` as `percentage` and `decimalPlace` respectively and `amount` as `100e18`, the `feeAmount` will come out be `(100e18 * 20) / 1` which means `2000e18` which is much larger than expected `20`.

To test this, I used this instead ⬇️ 
```
 if(_percentage.decimalPlaces == 0){
         divisor = 100;
    }else {
         divisor = 100 ** _percentage.decimalPlaces
    }
```
and it worked fine.

Reply:
Error mentioned is correct (100 was missed in the denominator)
Fix mentioned isn't correct 
Case: If _amount is 1000 and _percentage is {percentageNumber: 1234, decimalPlaces: 3}, it means (1.234% of 1000).
Acc. to fix mentioned
divisor = 100 ** _percentage.decimalPlaces = 1e6
then result = (1000 * 1234) / 1e6

ideal scenario
divisor = 10 ** _percentage.decimalPlaces = 1e3
then result = (1000 * 1234) / (1000 * 100)